### PR TITLE
Fix preview size

### PR DESCRIPTION
**Pull request by cinqmilleans**
**Resolved / Related Issues**
When the application window is resized and the PreviewPane is opened, the content size shrinks to keep the same size for the PreviewPane. This is the proper behavior.
But if we reduce too much, the content reaches the minimum possible size but the size of the PreviewPane is not reduced. This cannot be seen directly because the content of the PreviewPane adapts to the visible part. But the grid is too big and this can be seen in the InnerToolbar which no longer displays the elements on the right, which are displayed outside the available space. It also causes the PreviewPane to get the wrong size if you close it and reopen it. #9093

**Details of Changes**
Add an event to adapt the size of the PreviewPane when the size of the window is modified and the content has its minimum size.

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
Before/After

https://user-images.githubusercontent.com/46631671/167406957-d6ae5abc-b1c8-4903-a2ba-ab026bf0353e.mp4

https://user-images.githubusercontent.com/46631671/167405735-4f8e33e8-0c38-4499-9a89-c5abdc73800a.mp4

### CODE REVIEW

**Comment by yaichenbaum** [comment]
@cinqmilleans before merging this, it looks like the issue is already resolved in the `main` branch. The visual state is broken though so we should still take that fix.

**Comment by cinqmilleans** [comment]
@yair100 I'm not sure I understand your comment. This bug is not yet fixed in the main branch.

**Comment by yaichenbaum** [comment]
@cinqmilleans I can't reproduce it locally, can you make sure you have the latest commit?

**Comment by cinqmilleans** [comment]
@yair100 I just tested on the most up-to-date version of main, and there is the bug. To reproduce it, follow the instructions given here #9093. It takes a wide window, enlarge the PreviewPane almost to the maximum and reduce the window. You can see it in my "before" video at 11s.

**Comment by yair100** [comment]
> @yair100 I just tested on the most up-to-date version of main, and there is the bug. To reproduce it, follow the instructions given here #9093. It takes a wide window, enlarge the PreviewPane almost to the maximum and reduce the window. You can see it in my "before" video at 11s.

I still can't reproduce the issue, looking at the screen recording, it appears that you're testing an older version.

**Comment by cinqmilleans** [comment]
The video was made with the sideload version. The bug is still present in the latest version of main, as I just checked (commit 3df71f1f717ab67840bf10a04ef60cc3264caeab).

**Comment by yaichenbaum** [comment]
> The video was made with the sideload version. The bug is still present in the latest version of main, as I just checked (commit [3df71f1](https://github.com/files-community/Files/commit/3df71f1f717ab67840bf10a04ef60cc3264caeab)).

Can you upload a screen recording of the version in `main` so I can compare the results?

**Comment by cinqmilleans** [comment]
@yair100 Here.

https://user-images.githubusercontent.com/46631671/167614598-e980474e-e543-40fd-add1-1aa67d95daf2.mp4

**Comment by yaichenbaum** [comment]
@cinqmilleans I see the issue now, I thought you were talking about the column sizes changing in the details pane. I think this can all be solved by adjusting the xaml controls so we're going to hold off on this until after v2.3.

**Comment by cinqmilleans** [comment]
I fixed the merge, which was broken. I retested to validate the bug fix. This pr was pending release of version 2.3.

**Comment by yaichenbaum** [comment]
Instead of handling this in codebehind, is this something we can handle with a visual state?

File: src/Files.Uwp/Views/MainPage.xaml.cs
Comment (d2dyno1) [inline src/Files.Uwp/Views/MainPage.xaml.cs:391]: When you change the value of a setting, the whole cache is serialized. This will cause massive performance issues as well as storage strain

**Review by d2dyno1 — CHANGES_REQUESTED** [review]

**Review by d2dyno1 — APPROVED** [review]
looks good

**Comment by yaichenbaum** [comment]
> looks good

@d2dyno1 do you think we should use visual states here?

**Comment by d2dyno1** [comment]
I guess using visual states there would only make things harder

**Comment by yaichenbaum** [comment]
> I guess using visual states there would only make things harder

In what way? I find it easier when everything is done in xaml, it helps prevent surprises.

**Comment by d2dyno1** [comment]
@yair100 Settings are saved in the code behind if you take a look, won't be possible with xaml only iirc

**Review by yair100 — APPROVED** [review]

**Pull request merged**
